modules/core/src/lib/deck-picker.js

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {readPixelsToArray} from 'luma.gl';
+import {drawPickingBuffer} from './draw-layers';
+import getPixelRatio from '../utils/pixel-ratio';
+import assert from '../utils/assert';
+import {getClosestObject, getUniqueObjects} from '../utils/query-object';
+import {processPickInfo, getLayerPickingInfo} from '../utils/pick-info';
+
+export default class DeckPicker {
```

Comment:
What is the motivation of making it a class if all the methods are used as independent utility functions?

Reply:
deckPicker need store states like lastPickedInfo, pickingEvent, which were added in the second PR.